TS compilation issues

The ts compilation isn't picking up errors as it should. For example, adding

```
  "include": ["./src"],
```

to the invoices `tsconfig.json` results in

```
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(243,27): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'string'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(256,29): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'string'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(277,13): error TS2322: Type 'Invoice | undefined' is not assignable to type 'Invoice'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'Invoice'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(291,49): error TS2532: Object is possibly 'undefined'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(328,31): error TS2322: Type 'string | undefined' is not assignable to type 'string'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'string'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(333,15): error TS2322: Type 'Invoice | undefined' is not assignable to type 'Invoice'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'Invoice'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(347,51): error TS2532: Object is possibly 'undefined'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(354,3): error TS2322: Type '{ update: (variables: Partial<InboundShipment>, options?: MutateOptions<UpdateInboundShipmentMutation, unknown, Partial<InboundShipment>, { ...; }> | undefined) => Promise<...>; }' is not assignable to type '{ [k in KeyOfInvoice]: Invoice[k]; } & { update: (patch: Partial<Invoice>) => Promise<Promise<UpdateInboundShipmentMutation>>; }'.
@openmsupply-client/invoices:   Type '{ update: (variables: Partial<InboundShipment>, options?: MutateOptions<UpdateInboundShipmentMutation, unknown, Partial<InboundShipment>, { ...; }> | undefined) => Promise<...>; }' is not assignable to type '{ [k in KeyOfInvoice]: Invoice[k]; }'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(416,17): error TS2345: Argument of type 'OutboundShipmentSummaryItem | undefined' is not assignable to parameter of type 'InboundShipmentItem'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'InboundShipmentItem'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(447,29): error TS2345: Argument of type '{ items: never[]; status?: InvoiceNodeStatus | undefined; deleteLine?: ((line: InboundShipmentRow) => void) | undefined; deleteItem?: ((line: InboundShipmentItem) => void) | undefined; ... 18 more ...; verifiedDatetime?: Maybe<...> | undefined; }' is not assignable to parameter of type 'InboundShipment'.
@openmsupply-client/invoices:   Types of property 'status' are incompatible.
@openmsupply-client/invoices:     Type 'InvoiceNodeStatus | undefined' is not assignable to type 'InvoiceNodeStatus'.
@openmsupply-client/invoices:       Type 'undefined' is not assignable to type 'InvoiceNodeStatus'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(482,41): error TS2304: Cannot find name 'UseMutationResult'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(483,3): error TS2304: Cannot find name 'DeleteInboundShipmentLinesMutation'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(493,52): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'string'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(498,58): error TS2345: Argument of type '(old: Invoice | undefined) => { lines: InvoiceLine[]; status?: InvoiceNodeStatus | undefined; otherParty?: Name | undefined; pricing?: { ...; } | undefined; ... 15 more ...; verifiedDatetime?: Maybe<...> | undefined; }' is not assignable to parameter of type 'Updater<Invoice | undefined, Invoice>'.
@openmsupply-client/invoices:   Type '(old: Invoice | undefined) => { lines: InvoiceLine[]; status?: InvoiceNodeStatus | undefined; otherParty?: Name | undefined; pricing?: { ...; } | undefined; ... 15 more ...; verifiedDatetime?: Maybe<...> | undefined; }' is not assignable to type 'DataUpdateFunction<Invoice | undefined, Invoice>'.
@openmsupply-client/invoices:     Call signature return types '{ lines: InvoiceLine[]; status?: InvoiceNodeStatus | undefined; otherParty?: Name | undefined; pricing?: { totalAfterTax: number; } | undefined; ... 15 more ...; verifiedDatetime?: Maybe<...> | undefined; }' and 'Invoice' are incompatible.
@openmsupply-client/invoices:       The types of 'status' are incompatible between these types.
@openmsupply-client/invoices:         Type 'InvoiceNodeStatus | undefined' is not assignable to type 'InvoiceNodeStatus'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(500,16): error TS2532: Object is possibly 'undefined'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/api.ts(505,49): error TS2532: Object is possibly 'undefined'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/Footer.tsx(41,5): error TS2322: Type 'string' is not assignable to type 'null'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/Footer.tsx(44,5): error TS2322: Type 'Maybe<string> | undefined' is not assignable to type 'null'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'null'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/Footer.tsx(47,5): error TS2322: Type 'Maybe<string> | undefined' is not assignable to type 'null'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'null'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/Footer.tsx(50,5): error TS2322: Type 'Maybe<string> | undefined' is not assignable to type 'null'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'null'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/Footer.tsx(53,5): error TS2322: Type 'Maybe<string> | undefined' is not assignable to type 'null'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'null'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/Footer.tsx(79,23): error TS2339: Property 'update' does not exist on type 'InboundShipment'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/GeneralTab.tsx(60,26): error TS2344: Type 'InboundShipmentItem' does not satisfy the constraint 'RecordWithId & { canExpand: boolean; }'.
@openmsupply-client/invoices:   Property 'canExpand' is missing in type 'InboundShipmentItem' but required in type '{ canExpand: boolean; }'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/modals/InboundLineEdit/InboundLineEdit.tsx(143,32): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'string'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/modals/InboundLineEdit/InboundLineEdit.tsx(153,26): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'string'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/modals/InboundLineEdit/InboundLineEditForm.tsx(34,13): error TS2322: Type 'Item | null' is not assignable to type 'Item | undefined'.
@openmsupply-client/invoices:   Type 'null' is not assignable to type 'Item | undefined'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/modals/InboundLineEdit/TabTables.tsx(18,17): error TS2322: Type '<T extends RecordWithId>({ rowData, column, }: CellPropsWithUpdaterObject<T>) => React.ReactElement<CellPropsWithUpdaterObject<T>>' is not assignable to type 'JSXElementConstructor<CellProps<DraftInboundLine>> | undefined'.
@openmsupply-client/invoices:   Type '<T extends RecordWithId>({ rowData, column, }: CellPropsWithUpdaterObject<T>) => React.ReactElement<CellPropsWithUpdaterObject<T>>' is not assignable to type '(props: CellProps<DraftInboundLine>) => ReactElement<any, any> | null'.
@openmsupply-client/invoices:     Types of parameters '__0' and 'props' are incompatible.
@openmsupply-client/invoices:       Type 'CellProps<DraftInboundLine>' is not assignable to type 'CellPropsWithUpdaterObject<Pick<InvoiceLine, "id" | "batch" | "costPricePerPack" | "expiryDate" | "itemId" | "location" | "packSize" | "sellPricePerPack" | "numberOfPacks" | "invoiceId"> & { ...; }>'.
@openmsupply-client/invoices:         Types of property 'rowData' are incompatible.
@openmsupply-client/invoices:           Type 'DraftInboundLine' is not assignable to type 'DomainObjectWithUpdater<Pick<InvoiceLine, "id" | "batch" | "costPricePerPack" | "expiryDate" | "itemId" | "location" | "packSize" | "sellPricePerPack" | "numberOfPacks" | "invoiceId"> & { ...; }>'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/modals/InboundLineEdit/TabTables.tsx(22,9): error TS2322: Type '<T extends RecordWithId>({ rowData, column, }: CellPropsWithUpdaterObject<T>) => React.ReactElement<CellPropsWithUpdaterObject<T>>' is not assignable to type 'JSXElementConstructor<CellProps<DraftInboundLine>> | undefined'.
@openmsupply-client/invoices:   Type '<T extends RecordWithId>({ rowData, column, }: CellPropsWithUpdaterObject<T>) => React.ReactElement<CellPropsWithUpdaterObject<T>>' is not assignable to type '(props: CellProps<DraftInboundLine>) => ReactElement<any, any> | null'.
@openmsupply-client/invoices:     Types of parameters '__0' and 'props' are incompatible.
@openmsupply-client/invoices:       Type 'CellProps<DraftInboundLine>' is not assignable to type 'CellPropsWithUpdaterObject<Pick<InvoiceLine, "id" | "batch" | "costPricePerPack" | "expiryDate" | "itemId" | "location" | "packSize" | "sellPricePerPack" | "numberOfPacks" | "invoiceId"> & { ...; }>'.
@openmsupply-client/invoices:         Types of property 'rowData' are incompatible.
@openmsupply-client/invoices:           Type 'DraftInboundLine' is not assignable to type 'DomainObjectWithUpdater<Pick<InvoiceLine, "id" | "batch" | "costPricePerPack" | "expiryDate" | "itemId" | "location" | "packSize" | "sellPricePerPack" | "numberOfPacks" | "invoiceId"> & { ...; }>'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/modals/InboundLineEdit/TabTables.tsx(28,20): error TS2322: Type '<T extends RecordWithId>({ rowData, column, }: CellPropsWithUpdaterObject<T>) => React.ReactElement<CellPropsWithUpdaterObject<T>>' is not assignable to type 'JSXElementConstructor<CellProps<DraftInboundLine>> | undefined'.
@openmsupply-client/invoices:   Type '<T extends RecordWithId>({ rowData, column, }: CellPropsWithUpdaterObject<T>) => React.ReactElement<CellPropsWithUpdaterObject<T>>' is not assignable to type '(props: CellProps<DraftInboundLine>) => ReactElement<any, any> | null'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/modals/InboundLineEdit/TabTables.tsx(52,17): error TS2322: Type '<T extends RecordWithId>({ rowData, column, }: CellPropsWithUpdaterObject<T>) => React.ReactElement<CellPropsWithUpdaterObject<T>>' is not assignable to type 'JSXElementConstructor<CellProps<DraftInboundLine>> | undefined'.
@openmsupply-client/invoices:   Type '<T extends RecordWithId>({ rowData, column, }: CellPropsWithUpdaterObject<T>) => React.ReactElement<CellPropsWithUpdaterObject<T>>' is not assignable to type '(props: CellProps<DraftInboundLine>) => ReactElement<any, any> | null'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/modals/InboundLineEdit/TabTables.tsx(55,9): error TS2322: Type '<T extends RecordWithId>({ rowData, column, }: CellPropsWithUpdaterObject<T>) => React.ReactElement<CellPropsWithUpdaterObject<T>>' is not assignable to type 'JSXElementConstructor<CellProps<DraftInboundLine>> | undefined'.
@openmsupply-client/invoices:   Type '<T extends RecordWithId>({ rowData, column, }: CellPropsWithUpdaterObject<T>) => React.ReactElement<CellPropsWithUpdaterObject<T>>' is not assignable to type '(props: CellProps<DraftInboundLine>) => ReactElement<any, any> | null'.
@openmsupply-client/invoices:     Types of parameters '__0' and 'props' are incompatible.
@openmsupply-client/invoices:       Type 'CellProps<DraftInboundLine>' is not assignable to type 'CellPropsWithUpdaterObject<Pick<InvoiceLine, "id" | "batch" | "costPricePerPack" | "expiryDate" | "itemId" | "location" | "packSize" | "sellPricePerPack" | "numberOfPacks" | "invoiceId"> & { ...; }>'.
@openmsupply-client/invoices:         Types of property 'rowData' are incompatible.
@openmsupply-client/invoices:           Type 'DraftInboundLine' is not assignable to type 'DomainObjectWithUpdater<Pick<InvoiceLine, "id" | "batch" | "costPricePerPack" | "expiryDate" | "itemId" | "location" | "packSize" | "sellPricePerPack" | "numberOfPacks" | "invoiceId"> & { ...; }>'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/modals/InboundLineEdit/TabTables.tsx(59,9): error TS2322: Type '<T extends RecordWithId>({ rowData, column, }: CellPropsWithUpdaterObject<T>) => React.ReactElement<CellPropsWithUpdaterObject<T>>' is not assignable to type 'JSXElementConstructor<CellProps<DraftInboundLine>> | undefined'.
@openmsupply-client/invoices:   Type '<T extends RecordWithId>({ rowData, column, }: CellPropsWithUpdaterObject<T>) => React.ReactElement<CellPropsWithUpdaterObject<T>>' is not assignable to type '(props: CellProps<DraftInboundLine>) => ReactElement<any, any> | null'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/Toolbar.tsx(39,26): error TS2532: Object is possibly 'undefined'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/Toolbar.tsx(41,15): error TS2339: Property 'batches' does not exist on type 'OutboundShipmentSummaryItem | undefined'.
@openmsupply-client/invoices: src/InboundShipment/DetailView/Toolbar.tsx(43,12): error TS2345: Argument of type '({ id }: { id: any; }) => any' is not assignable to parameter of type '(value: unknown, index: number, array: unknown[]) => any'.
@openmsupply-client/invoices:   Types of parameters '__0' and 'value' are incompatible.
@openmsupply-client/invoices:     Type 'unknown' is not assignable to type '{ id: any; }'.
@openmsupply-client/invoices: src/InboundShipment/ListView/api.ts(21,5): error TS2322: Type 'InvoiceNodeStatus' is not assignable to type 'Maybe<UpdateOutboundShipmentStatusInput> | undefined'.
@openmsupply-client/invoices: src/InboundShipment/ListView/api.ts(119,7): error TS2322: Type 'UpdateOutboundShipmentInput' is not assignable to type 'UpdateInboundShipmentInput'.
@openmsupply-client/invoices:   Types of property 'status' are incompatible.
@openmsupply-client/invoices:     Type 'Maybe<UpdateOutboundShipmentStatusInput> | undefined' is not assignable to type 'Maybe<UpdateInboundShipmentStatusInput> | undefined'.
@openmsupply-client/invoices:       Type 'UpdateOutboundShipmentStatusInput.Allocated' is not assignable to type 'Maybe<UpdateInboundShipmentStatusInput> | undefined'.
@openmsupply-client/invoices: src/InboundShipment/ListView/ListView.tsx(60,55): error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'InvoiceNodeStatus'.
@openmsupply-client/invoices: src/InboundShipment/ListView/Toolbar.tsx(41,34): error TS2769: No overload matches this call.
@openmsupply-client/invoices:   Overload 1 of 2, '(key?: LocaleKey | LocaleKey[] | undefined, options?: string | TOptions<Record<string, unknown>> | undefined): string', gave the following error.
@openmsupply-client/invoices:     Argument of type '"messages.deleted-invoices"' is not assignable to parameter of type 'LocaleKey | LocaleKey[] | undefined'.
@openmsupply-client/invoices:   Overload 2 of 2, '(key?: LocaleKey | LocaleKey[] | undefined, defaultValue?: string | undefined, options?: string | TOptions<Record<string, unknown>> | undefined): string', gave the following error.
@openmsupply-client/invoices:     Argument of type '"messages.deleted-invoices"' is not assignable to parameter of type 'LocaleKey | LocaleKey[] | undefined'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/DetailView.tsx(125,26): error TS2344: Type 'OutboundShipmentSummaryItem' does not satisfy the constraint 'RecordWithId & { canExpand: boolean; }'.
@openmsupply-client/invoices:   Type 'OutboundShipmentSummaryItem' is not assignable to type '{ canExpand: boolean; }'.
@openmsupply-client/invoices:     Types of property 'canExpand' are incompatible.
@openmsupply-client/invoices:       Type 'boolean | undefined' is not assignable to type 'boolean'.
@openmsupply-client/invoices:         Type 'undefined' is not assignable to type 'boolean'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/Footer.tsx(43,5): error TS2322: Type 'string' is not assignable to type 'null'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/Footer.tsx(46,5): error TS2322: Type 'Maybe<string> | undefined' is not assignable to type 'null'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'null'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/Footer.tsx(49,5): error TS2322: Type 'Maybe<string> | undefined' is not assignable to type 'null'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'null'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/Footer.tsx(52,5): error TS2322: Type 'Maybe<string> | undefined' is not assignable to type 'null'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'null'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/Footer.tsx(55,5): error TS2322: Type 'Maybe<string> | undefined' is not assignable to type 'null'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'null'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/Footer.tsx(58,5): error TS2322: Type 'Maybe<string> | undefined' is not assignable to type 'null'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'null'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx(214,55): error TS2345: Argument of type '{ onClose: () => void; }' is not assignable to parameter of type 'DialogProps'.
@openmsupply-client/invoices:   Property 'isOpen' is missing in type '{ onClose: () => void; }' but required in type 'DialogProps'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/reducer.ts(73,5): error TS2322: Type 'Maybe<string> | undefined' is not assignable to type 'string | undefined'.
@openmsupply-client/invoices:   Type 'null' is not assignable to type 'string | undefined'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/reducer.ts(143,5): error TS2322: Type 'Maybe<string> | undefined' is not assignable to type 'string | undefined'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/reducer.ts(225,17): error TS2322: Type 'number | undefined' is not assignable to type 'number'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'number'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/reducer.ts(226,17): error TS2322: Type 'number | undefined' is not assignable to type 'number'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'number'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/reducer.ts(234,15): error TS2322: Type 'number | undefined' is not assignable to type 'number'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'number'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/reducer.ts(235,15): error TS2322: Type 'number | undefined' is not assignable to type 'number'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'number'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/reducer.ts(354,15): error TS2322: Type 'number | undefined' is not assignable to type 'number'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'number'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/reducer.ts(355,15): error TS2322: Type 'number | undefined' is not assignable to type 'number'.
@openmsupply-client/invoices:   Type 'undefined' is not assignable to type 'number'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/tabs/GeneralTab.tsx(97,24): error TS2345: Argument of type '{ outboundShipment: boolean; inboundShipment?: boolean | undefined; }' is not assignable to parameter of type 'GroupByItem'.
@openmsupply-client/invoices:   Types of property 'inboundShipment' are incompatible.
@openmsupply-client/invoices:     Type 'boolean | undefined' is not assignable to type 'boolean'.
@openmsupply-client/invoices:       Type 'undefined' is not assignable to type 'boolean'.
@openmsupply-client/invoices: src/OutboundShipment/DetailView/tabs/GeneralTab.tsx(121,13): error TS2322: Type 'Maybe<string> | undefined' is not assignable to type 'string | undefined'.
@openmsupply-client/invoices: src/OutboundShipment/ListView/api.ts(104,5): error TS2322: Type 'InvoiceNodeStatus' is not assignable to type 'Maybe<UpdateOutboundShipmentStatusInput> | undefined'.
@openmsupply-client/invoices: src/OutboundShipment/ListView/api.ts(152,23): error TS2345: Argument of type 'SortBy<InvoiceRow>' is not assignable to parameter of type 'SortBy<OutboundShipment>'.
@openmsupply-client/invoices:   Type 'OutboundShipment' is not assignable to type 'InvoiceRow'.
@openmsupply-client/invoices:     Types of property 'pricing' are incompatible.
@openmsupply-client/invoices:       Type '{ totalAfterTax: number; }' is missing the following properties from type 'InvoicePricingNode': serviceTotalAfterTax, serviceTotalBeforeTax, stockTotalAfterTax, stockTotalBeforeTax, totalBeforeTax
@openmsupply-client/invoices: src/OutboundShipment/ListView/api.ts(153,25): error TS2345: Argument of type 'SortBy<InvoiceRow>' is not assignable to parameter of type 'SortBy<OutboundShipment>'.
@openmsupply-client/invoices: error Command failed with exit code 2.
@openmsupply-client/invoices: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

--- a/packages/common/src/hooks/useDialog/useDialog.tsx
+++ b/packages/common/src/hooks/useDialog/useDialog.tsx
@@ -25,7 +25,7 @@ export interface ModalProps {
 }
 export interface DialogProps {
   onClose?: () => void;
-  isOpen: boolean;
+  isOpen?: boolean;
 }
 
 interface DialogState {

--- a/packages/common/src/intl/locales/en/replenishment.json
+++ b/packages/common/src/intl/locales/en/replenishment.json
@@ -6,5 +6,6 @@
   "message.select-rows-to-delete": "Select rows to delete",
   "message.cant-delete-requisitions": "Can only delete requisitions with a status of 'Draft'",
   "message.deleted-requisitions": "Deleted {{number}} requisitions",
-  "message.deleted-lines": "Deleted {{number}} lines"
+  "message.deleted-lines": "Deleted {{number}} lines",
+  "messages.deleted-invoices": "Deleted {{number}} invoices"
 }

--- a/packages/common/src/localStorage/LocalStorage.ts
+++ b/packages/common/src/localStorage/LocalStorage.ts
@@ -46,6 +46,7 @@ class LocalStorage {
     if (existingValue === value) return;
 
     const stringified = JSON.stringify(value);
+
     localStorage.setItem(this.createStorageKey(key), stringified);
 
     this.listeners.forEach(listener => listener(key, value));

--- a/packages/common/src/localStorage/keys.ts
+++ b/packages/common/src/localStorage/keys.ts
@@ -1,8 +1,8 @@
 import { SupportedLocales } from '@common/intl';
 
 type GroupByItem = {
-  outboundShipment: boolean;
-  inboundShipment: boolean;
+  outboundShipment?: boolean;
+  inboundShipment?: boolean;
 };
 export type LocalStorageRecord = {
   '/appdrawer/open': boolean;

--- a/packages/common/src/ui/layout/tables/columns/RowExpand/RowExpand.tsx
+++ b/packages/common/src/ui/layout/tables/columns/RowExpand/RowExpand.tsx
@@ -13,7 +13,7 @@ type RowExpandLabels = {
 };
 
 export const getRowExpandColumn = <
-  T extends DomainObject & { canExpand: boolean }
+  T extends DomainObject & { canExpand?: boolean }
 >(
   labels?: RowExpandLabels
 ): ColumnDefinition<T> => ({

--- a/packages/common/src/ui/layout/tables/components/Cells/CurrencyInputCell/CurrencyInputCell.tsx
+++ b/packages/common/src/ui/layout/tables/components/Cells/CurrencyInputCell/CurrencyInputCell.tsx
@@ -4,20 +4,10 @@ import { CurrencyInput } from '@common/components';
 import { DomainObject } from '@common/types';
 import { useBufferState, useDebounceCallback } from '@common/hooks';
 
-type RowData<T> = T & DomainObject;
-
-type DomainObjectWithUpdater<T> = RowData<T> & {
-  update: (patch: Partial<RowData<T>>) => void;
-};
-
-type CellPropsWithUpdaterObject<T> = CellProps<DomainObjectWithUpdater<T>>;
-
 export const CurrencyInputCell = <T extends DomainObject>({
   rowData,
   column,
-}: CellPropsWithUpdaterObject<T>): React.ReactElement<
-  CellPropsWithUpdaterObject<T>
-> => {
+}: CellProps<T>): React.ReactElement<CellProps<T>> => {
   const [buffer, setBuffer] = useBufferState(Number(column.accessor(rowData)));
 
   const updater = useDebounceCallback(column.setter, [rowData], 250);

--- a/packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NumberInputCell.tsx
+++ b/packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NumberInputCell.tsx
@@ -4,20 +4,10 @@ import { BasicTextInput } from '@common/components';
 import { DomainObject } from '@common/types';
 import { useBufferState, useDebounceCallback } from '@common/hooks';
 
-type RowData<T> = T & DomainObject;
-
-type DomainObjectWithUpdater<T> = RowData<T> & {
-  update: (patch: Partial<RowData<T>>) => void;
-};
-
-type CellPropsWithUpdaterObject<T> = CellProps<DomainObjectWithUpdater<T>>;
-
 export const NumberInputCell = <T extends DomainObject>({
   rowData,
   column,
-}: CellPropsWithUpdaterObject<T>): React.ReactElement<
-  CellPropsWithUpdaterObject<T>
-> => {
+}: CellProps<T>): React.ReactElement<CellProps<T>> => {
   const [buffer, setBuffer] = useBufferState(column.accessor(rowData));
   const updater = useDebounceCallback(column.setter, [rowData], 250);
 

--- a/packages/common/src/ui/layout/tables/components/Cells/TextInputCell/TextInputCell.tsx
+++ b/packages/common/src/ui/layout/tables/components/Cells/TextInputCell/TextInputCell.tsx
@@ -4,20 +4,10 @@ import { BasicTextInput } from '@common/components';
 import { DomainObject } from '@common/types';
 import { useBufferState, useDebounceCallback } from '@common/hooks';
 
-type RowData<T> = T & DomainObject;
-
-type DomainObjectWithUpdater<T> = RowData<T> & {
-  update: (patch: Partial<RowData<T>>) => void;
-};
-
-type CellPropsWithUpdaterObject<T> = CellProps<DomainObjectWithUpdater<T>>;
-
 export const TextInputCell = <T extends DomainObject>({
   rowData,
   column,
-}: CellPropsWithUpdaterObject<T>): React.ReactElement<
-  CellPropsWithUpdaterObject<T>
-> => {
+}: CellProps<T>): React.ReactElement<CellProps<T>> => {
   const [buffer, setBuffer] = useBufferState(column.accessor(rowData));
   const updater = useDebounceCallback(column.setter, [rowData], 500);
 

--- a/packages/invoices/src/InboundShipment/DetailView/DetailView.tsx
+++ b/packages/invoices/src/InboundShipment/DetailView/DetailView.tsx
@@ -57,7 +57,7 @@ export const DetailView: FC = () => {
         onAddItem={onAddItem}
       />
 
-      <Toolbar draft={draft} update={updateInvoice} />
+      <Toolbar draft={draft} />
 
       <GeneralTab onRowClick={onRowClick} />
 

--- a/packages/invoices/src/InboundShipment/DetailView/Footer.tsx
+++ b/packages/invoices/src/InboundShipment/DetailView/Footer.tsx
@@ -13,10 +13,13 @@ import React, { FC } from 'react';
 import {
   getNextInboundStatusButtonTranslation,
   getStatusTranslator,
-  isInboundEditable,
   inboundStatuses,
 } from '../../utils';
 import { InboundShipment } from '../../types';
+import {
+  useInboundFields,
+  useIsInboundEditable,
+} from 'packages/invoices/src/InboundShipment/DetailView/api';
 
 interface InboundDetailFooterProps {
   draft: InboundShipment;
@@ -25,7 +28,7 @@ interface InboundDetailFooterProps {
 
 const createStatusLog = (draft: InboundShipment) => {
   const statusIdx = inboundStatuses.findIndex(s => draft.status === s);
-  const statusLog = {
+  const statusLog: Record<InvoiceNodeStatus, null | string | undefined> = {
     [InvoiceNodeStatus.New]: null,
     [InvoiceNodeStatus.Picked]: null,
     [InvoiceNodeStatus.Shipped]: null,
@@ -59,11 +62,14 @@ const createStatusLog = (draft: InboundShipment) => {
 export const Footer: FC<InboundDetailFooterProps> = ({ draft, save }) => {
   const t = useTranslation('common');
   const { success } = useNotification();
+  const { onHold, status, update } = useInboundFields(['onHold', 'status']);
+  const isEditable = useIsInboundEditable();
 
   return (
     <AppFooterPortal
       Content={
-        !!draft && (
+        !!onHold &&
+        !!status && (
           <Box
             gap={2}
             display="flex"
@@ -72,11 +78,11 @@ export const Footer: FC<InboundDetailFooterProps> = ({ draft, save }) => {
             height={64}
           >
             <ToggleButton
-              disabled={!isInboundEditable(draft)}
-              value={!!draft.onHold}
-              selected={!!draft.onHold}
+              disabled={!isEditable}
+              value={onHold}
+              selected={onHold}
               onClick={(_, value) => {
-                draft.update?.('onHold', !value);
+                update?.({ onHold: !value });
               }}
               label={t('label.hold')}
             />
@@ -88,16 +94,14 @@ export const Footer: FC<InboundDetailFooterProps> = ({ draft, save }) => {
             />
 
             <Box flex={1} display="flex" justifyContent="flex-end" gap={2}>
-              {isInboundEditable(draft) && (
+              {!isEditable && (
                 <>
                   <ButtonWithIcon
                     shrinkThreshold="lg"
-                    disabled={draft.onHold}
+                    disabled={onHold}
                     Icon={<ArrowRightIcon />}
                     label={t('button.save-and-confirm-status', {
-                      status: t(
-                        getNextInboundStatusButtonTranslation(draft.status)
-                      ),
+                      status: t(getNextInboundStatusButtonTranslation(status)),
                     })}
                     sx={{ fontSize: '12px' }}
                     variant="contained"

--- a/packages/invoices/src/InboundShipment/DetailView/Toolbar.tsx
+++ b/packages/invoices/src/InboundShipment/DetailView/Toolbar.tsx
@@ -14,12 +14,11 @@ import {
 } from '@openmsupply-client/common';
 import { NameSearchInput } from '@openmsupply-client/system/src/Name';
 import { useDeleteInboundLine, useInboundFields, useInboundItems } from './api';
-import { InboundShipment } from '../../types';
+import { InboundShipment, InboundShipmentItem } from '../../types';
 import { isInboundEditable } from '../../utils';
 
 interface ToolbarProps {
   draft: InboundShipment;
-  update: (patch: Partial<InboundShipment>) => Promise<InboundShipment>;
 }
 
 export const Toolbar: FC<ToolbarProps> = ({ draft }) => {
@@ -30,14 +29,18 @@ export const Toolbar: FC<ToolbarProps> = ({ draft }) => {
     'theirReference',
   ]);
 
+  if (!data) return null;
+
   const t = useTranslation(['replenishment', 'common']);
   const { success, info } = useNotification();
 
   const { selectedRows } = useTableStore(state => ({
-    selectedRows: Object.keys(state.rowState)
-      .filter(id => state.rowState[id]?.isSelected)
-      .map(selectedId => data.find(({ id }) => selectedId === id))
-      .filter(Boolean)
+    selectedRows: (
+      Object.keys(state.rowState)
+        .filter(id => state.rowState[id]?.isSelected)
+        .map(selectedId => data.find(({ id }) => selectedId === id))
+        .filter(Boolean) as InboundShipmentItem[]
+    )
       .map(({ batches }) => Object.values(batches))
       .flat()
       .map(({ id }) => id),

--- a/packages/invoices/src/InboundShipment/DetailView/api.ts
+++ b/packages/invoices/src/InboundShipment/DetailView/api.ts
@@ -2,6 +2,8 @@ import { toItem } from './DetailView';
 import { DraftInboundLine } from './modals/InboundLineEdit/InboundLineEdit';
 import { useCallback } from 'react';
 import {
+  UseMutationResult,
+  DeleteInboundShipmentLinesMutation,
   InvoiceNodeStatus,
   UpdateInboundShipmentMutation,
   useParams,
@@ -235,7 +237,7 @@ export const getInboundShipmentDetailViewApi = (
 });
 
 export const useInboundShipment = (): UseQueryResult<Invoice, unknown> => {
-  const { id } = useParams();
+  const { id = '' } = useParams();
 
   const { api } = useOmSupplyApi();
   const queries = getInboundShipmentDetailViewApi(api);
@@ -247,7 +249,7 @@ export const useInboundShipment = (): UseQueryResult<Invoice, unknown> => {
 export const useInboundShipmentSelector = <T = Invoice>(
   select?: (data: Invoice) => T
 ): UseQueryResult<T, unknown> => {
-  const { id } = useParams();
+  const { id = '' } = useParams();
   const { api } = useOmSupplyApi();
   const queries = getInboundShipmentDetailViewApi(api);
   return useQuery(
@@ -274,21 +276,20 @@ const useOptimisticInboundUpdate = () => {
     onMutate: async (patch: Partial<InboundShipment>) => {
       await queryClient.cancelQueries(['invoice', id]);
 
-      const previousInbound: Invoice = queryClient.getQueryData([
-        'invoice',
-        id,
-      ]);
+      const previous = queryClient.getQueryData<Invoice>(['invoice', id]);
 
-      queryClient.setQueryData(['invoice', id], {
-        ...previousInbound,
-        ...patch,
-      });
+      if (previous) {
+        queryClient.setQueryData<Invoice>(['invoice', id], {
+          ...previous,
+          ...patch,
+        });
+      }
 
-      return { previousInbound, patch };
+      return { previous, patch };
     },
     onSettled: () => queryClient.invalidateQueries(['invoice', id]),
     onError: (_, __, context) => {
-      queryClient.setQueryData(['invoice', id], context.previousInbound);
+      queryClient.setQueryData(['invoice', id], context?.previous);
     },
   });
 };
@@ -302,7 +303,7 @@ export const useInboundFields = <KeyOfInvoice extends keyof Invoice>(
   ) => Promise<Promise<UpdateInboundShipmentMutation>>;
 } => {
   const queryClient = useQueryClient();
-  const { id } = useParams();
+  const { id = '' } = useParams();
   const { api } = useOmSupplyApi();
   const select = useCallback(
     (invoice: Invoice) => {
@@ -330,28 +331,32 @@ export const useInboundFields = <KeyOfInvoice extends keyof Invoice>(
       onMutate: async (patch: Partial<InboundShipment>) => {
         await queryClient.cancelQueries(['invoice', id]);
 
-        const previousInbound: Invoice = queryClient.getQueryData([
-          'invoice',
-          id,
-        ]);
+        const previous = queryClient.getQueryData<Invoice>(['invoice', id]);
 
-        queryClient.setQueryData(['invoice', id], {
-          ...previousInbound,
-          ...patch,
-        });
+        if (previous) {
+          queryClient.setQueryData<Invoice>(['invoice', id], {
+            ...previous,
+            ...patch,
+          });
+        }
 
-        return { previousInbound, patch };
+        return { previous, patch };
       },
       onSettled: () => queryClient.invalidateQueries(['invoice', id]),
       onError: (_, __, context) => {
-        queryClient.setQueryData(['invoice', id], context.previousInbound);
+        queryClient.setQueryData(['invoice', id], context?.previous);
       },
     }
   );
 
   const update = useDebounceCallback(mutateAsync, [], timeout);
 
-  return { ...data, update };
+  // When data is undefined, just return an empty object instead of undefined.
+  // This allows the caller to use, for example, const { comment } = useInboundFields('comment')
+  // and the comment is undefined when the invoice has not been fetched yet.
+  const returnVal = data ?? ({} as { [k in KeyOfInvoice]: Invoice[k] });
+
+  return { ...returnVal, update };
 };
 
 export const useIsInboundEditable = (): boolean => {
@@ -407,11 +412,9 @@ export const useNextItem = (currentItemId: string) => {
   const { data } = useInboundItems();
 
   if (!data) return null;
-
-  const currentIndex = data?.findIndex(
-    ({ itemId }) => itemId === currentItemId
-  );
-  const nextItem = data[(currentIndex + 1) % data.length];
+  const currentIndex = data.findIndex(({ itemId }) => itemId === currentItemId);
+  const nextItem = data?.[(currentIndex + 1) % data.length];
+  if (!nextItem) return null;
 
   return toItem(nextItem);
 };
@@ -444,7 +447,8 @@ export const useDraftInbound = () => {
   );
 
   const updateInvoice = async (patch: Partial<InboundShipment>) => {
-    return optimisticUpdate({ ...data, ...patch, items: [] });
+    if (!data) return;
+    else return optimisticUpdate({ ...data, ...patch, items: [] });
   };
 
   const upsertItem = async (item: OutboundShipmentSummaryItem) => {
@@ -483,26 +487,33 @@ export const useDeleteInboundLine = (): UseMutationResult<
   DeleteInboundShipmentLinesMutation,
   unknown,
   string[],
-  { previous: Invoice; ids: string[] }
+  { previous?: Invoice; ids: string[] }
 > => {
   // TODO: Shouldn't need to get the invoice ID here from the params as the mutation
   // input object should not require the invoice ID. Waiting for an API change.
-  const { id } = useParams();
+  const { id = '' } = useParams();
   const queryClient = useQueryClient();
   const { api } = useOmSupplyApi();
   const mutation = getDeleteInboundLinesQuery(api, id);
   return useMutation(mutation, {
     onMutate: async (ids: string[]) => {
       await queryClient.cancelQueries(['invoice', id]);
+
       const previous = queryClient.getQueryData<Invoice>(['invoice', id]);
-      queryClient.setQueryData<Invoice>(['invoice', id], old => ({
-        ...old,
-        lines: old.lines.filter(({ id: lineId }) => !ids.includes(lineId)),
-      }));
+
+      if (previous) {
+        queryClient.setQueryData<Invoice>(['invoice', id], {
+          ...previous,
+          lines: previous.lines.filter(
+            ({ id: lineId }) => !ids.includes(lineId)
+          ),
+        });
+      }
+
       return { previous, ids };
     },
     onError: (_, __, context) => {
-      queryClient.setQueryData(['invoice', id], context.previous);
+      queryClient.setQueryData(['invoice', id], context?.previous);
     },
     onSettled: () => {
       queryClient.invalidateQueries(['invoice', id]);

--- a/packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/InboundLineEdit.tsx
+++ b/packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/InboundLineEdit.tsx
@@ -140,7 +140,7 @@ export const InboundLineEdit: FC<InboundLineEditProps> = ({
   const t = useTranslation('distribution');
   const { error } = useNotification();
   const [currentItem, setCurrentItem] = useState<Item | null>(item);
-  const nextItem = useNextItem(currentItem?.id);
+  const nextItem = useNextItem(currentItem?.id ?? '');
   const isMediumScreen = useIsMediumScreen();
   const [currentTab, setCurrentTab] = useState<Tabs>(Tabs.Batch);
   const { Modal } = useDialog({ isOpen, onClose });
@@ -150,7 +150,7 @@ export const InboundLineEdit: FC<InboundLineEditProps> = ({
   }, [item]);
 
   const { draftLines, addDraftLine, updateDraftLine, isLoading, saveLines } =
-    useDraftInboundLines(currentItem?.id);
+    useDraftInboundLines(currentItem?.id ?? '');
 
   return (
     <Modal

--- a/packages/invoices/src/InboundShipment/ListView/ListView.tsx
+++ b/packages/invoices/src/InboundShipment/ListView/ListView.tsx
@@ -8,6 +8,7 @@ import {
   TableProvider,
   createTableStore,
   // Color,
+  InvoiceNodeStatus,
   useOmSupplyApi,
   useNotification,
   generateUUID,
@@ -57,7 +58,8 @@ export const InboundListView: FC = () => {
       [
         'status',
         {
-          formatter: status => getStatusTranslator(t)(status),
+          formatter: status =>
+            getStatusTranslator(t)(status as InvoiceNodeStatus),
         },
       ],
       'invoiceNumber',

--- a/packages/invoices/src/InboundShipment/ListView/api.ts
+++ b/packages/invoices/src/InboundShipment/ListView/api.ts
@@ -1,6 +1,6 @@
 import {
-  InvoiceNodeStatus,
-  UpdateOutboundShipmentInput,
+  UpdateInboundShipmentStatusInput,
+  UpdateInboundShipmentInput,
   InvoicesQuery,
   SortBy,
   ListApi,
@@ -13,12 +13,12 @@ import { Invoice, InvoiceRow } from '../../types';
 
 const invoiceToInput = (
   patch: Partial<Invoice> & { id: string }
-): UpdateOutboundShipmentInput => {
+): UpdateInboundShipmentInput => {
   return {
     id: patch.id,
     // color: patch.color,
     comment: patch.comment,
-    status: patch.status as InvoiceNodeStatus,
+    status: patch.status as unknown as UpdateInboundShipmentStatusInput,
     onHold: patch.onHold,
     otherPartyId: patch.otherParty?.id,
     theirReference: patch.theirReference,

--- a/packages/invoices/src/OutboundShipment/DetailView/Footer.tsx
+++ b/packages/invoices/src/OutboundShipment/DetailView/Footer.tsx
@@ -30,7 +30,7 @@ interface OutboundDetailFooterProps {
 const createStatusLog = (draft: OutboundShipment) => {
   const statusIdx = outboundStatuses.findIndex(s => draft.status === s);
 
-  const statusLog = {
+  const statusLog: Record<InvoiceNodeStatus, null | undefined | string> = {
     [InvoiceNodeStatus.New]: null,
     [InvoiceNodeStatus.Allocated]: null,
     [InvoiceNodeStatus.Picked]: null,

--- a/packages/invoices/src/OutboundShipment/DetailView/reducer.ts
+++ b/packages/invoices/src/OutboundShipment/DetailView/reducer.ts
@@ -51,7 +51,15 @@ const getExistingLine = (
 
 export const recalculateSummary = (
   summaryItem: OutboundShipmentSummaryItem
-): Partial<OutboundShipmentSummaryItem> => {
+): {
+  unitQuantity: number;
+  numberOfPacks: number;
+  locationName?: string | null;
+  batch?: string | null;
+  expiryDate?: string | null;
+  sellPricePerPack?: number;
+  packSize?: number;
+} => {
   const batches = Object.values<OutboundShipmentRow>(summaryItem.batches);
   const unitQuantity = batches.reduce(getUnitQuantity, 0);
   const numberOfPacks = batches.reduce(getSumOfKeyReducer('numberOfPacks'), 0);

--- a/packages/invoices/src/OutboundShipment/ListView/api.ts
+++ b/packages/invoices/src/OutboundShipment/ListView/api.ts
@@ -1,5 +1,4 @@
 import {
-  InvoiceNodeStatus,
   UpdateOutboundShipmentInput,
   InvoicesQuery,
   SortBy,
@@ -8,8 +7,9 @@ import {
   InvoicesQueryVariables,
   InvoicePriceResponse,
   OmSupplyApi,
+  UpdateOutboundShipmentStatusInput,
 } from '@openmsupply-client/common';
-import { OutboundShipment, Invoice, InvoiceRow } from '../../types';
+import { Invoice, InvoiceRow } from '../../types';
 
 const pricingGuard = (pricing: InvoicePriceResponse) => {
   if (pricing.__typename === 'InvoicePricingNode') {
@@ -101,16 +101,14 @@ const invoiceToInput = (
     id: patch.id,
     // color: patch.color,
     comment: patch.comment,
-    status: patch.status as InvoiceNodeStatus,
+    status: patch.status as unknown as UpdateOutboundShipmentStatusInput,
     onHold: patch.onHold,
     otherPartyId: patch.otherParty?.id,
     theirReference: patch.theirReference,
   };
 };
 
-const getSortKey = (
-  sortBy: SortBy<OutboundShipment>
-): InvoiceSortFieldInput => {
+const getSortKey = (sortBy: SortBy<InvoiceRow>): InvoiceSortFieldInput => {
   switch (sortBy.key) {
     // case 'allocatedDatetime': {
     //   return InvoiceSortFieldInput.ConfirmDatetime;
@@ -138,7 +136,7 @@ const getSortKey = (
   }
 };
 
-const getSortDesc = (sortBy: SortBy<OutboundShipment>): boolean => {
+const getSortDesc = (sortBy: SortBy<InvoiceRow>): boolean => {
   return !!sortBy.isDesc;
 };
 

--- a/packages/invoices/src/types.ts
+++ b/packages/invoices/src/types.ts
@@ -134,7 +134,7 @@ export type OutboundShipmentSummaryItem = {
   note?: string | null;
   isDeleted?: boolean;
   canExpand?: boolean;
-  expiryDate?: string;
+  expiryDate?: string | null;
   lineTotal?: number;
 };
 

--- a/packages/invoices/tsconfig.json
+++ b/packages/invoices/tsconfig.json
@@ -1,5 +1,6 @@
 {
   "extends": "../../tsconfig.json",
+  "include": ["./src"],
   "compilerOptions": {
     "rootDir": "./src"
   }

--- a/packages/system/src/Item/Components/ItemSearchInput/ItemSearchInput.tsx
+++ b/packages/system/src/Item/Components/ItemSearchInput/ItemSearchInput.tsx
@@ -37,7 +37,7 @@ const getOptionRenderer =
 
 interface ItemSearchInputProps {
   onChange: (item: Item | null) => void;
-  currentItem?: Item;
+  currentItem?: Item | null;
   currentItemName?: string;
   disabled?: boolean;
   extraFilter?: (item: Item) => boolean;

